h2o-persist-gcs/src/main/java/water/persist/PersistGcs.java

```diff
@@ -0,0 +1,161 @@
+package water.persist;
+
+import com.google.cloud.ReadChannel;
+import com.google.cloud.storage.*;
+import water.H2O;
+import water.Key;
+import water.Value;
+import water.api.FSIOException;
+import water.fvec.GcsFileVec;
+import water.util.Log;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Persistence backend for GCS
+ */
+public final class PersistGcs extends Persist {
+
+  private static final Storage storage = StorageOptions.getDefaultInstance().getService();
+
+  @Override
+  public byte[] load(Value v) throws IOException {
+    final BlobId blobId = GcsBlob.of(v._key).getBlobId();
+    Log.info("Loading: " + blobId.toString());
+    return storage.get(blobId).getContent();
+  }
+
+  @Override
+  public Key uriToKey(URI uri) throws IOException {
+    final GcsBlob blob = GcsBlob.of(uri);
+    final Long contentSize = storage.get(blob.getBlobId()).getSize();
```

Comment:
In this play, we may get a com.google.cloud.storage.StorageException if a file is accessed without proper access rights (401 Unauthorized).

Reply:
The exception is informative enough, without exposing any sensitive information to logs.  No need to wrap.